cmd/k8s-operator/operator.go

```diff
@@ -322,3 +323,10 @@ func serviceHandler(_ context.Context, o client.Object) []reconcile.Request {
 	}
 
 }
+
+// isMagicDNSName reports whether name is a full tailnet node FQDN (with or
+// without final dot).
+func isMagicDNSName(name string) bool {
+	validMagicDNSName := regexp.MustCompile(`^[a-zA-Z0-9-]+\.[a-zA-Z0-9-]+\.ts\.net\.?$`)
```

Comment:
I am not sure if we really want the check to be this strict. If at some point we support BYOD or 3rd level hostnames (foo.bar.tail-scale.ts.net) we'll need to update this, and ask users to upgrade the operator. I don't feel too strongly about this, but perhaps `^[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$` would be enough.

Reply:
> If at some point we support BYOD or 3rd level hostnames (foo.bar.tail-scale.ts.net) we'll need to update this, and ask users to upgrade the operator. I don't feel too strongly about this, but perhaps ^[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$ would be enough.

That's really non-specific though, it wouldn't catch for example if users passed <node-name>.<tailnet-name> which is a mistake they could quite likely make. I think it's not unreasonable to ask users to upgrade operator if in the future we actually implement BYOD _and_ they want to use that.